ports/rp2/machine_spi.c

```diff
@@ -172,7 +180,12 @@ mp_obj_t machine_spi_make_new(const mp_obj_type_t *type, size_t n_args, size_t n
         }
         self->mosi = mosi;
     }
-    if (args[ARG_miso].u_obj != mp_const_none) {
+
+    if (args[ARG_miso].u_obj == MP_ROM_INT(-1)) {
```

Comment:
At this state of the commit series, this if clause is wrong. If the argument is not present, the content of the static SPI object must not be changed.